NuRadioReco/modules/triggerTimeAdjuster.py

```diff
@@ -97,68 +100,74 @@ def run(self, event, station, detector, mode='sim_to_data'):
-                        2 * np.ceil( # this should ensure that 1) the number of samples is even and 2) resampling to the detector sampling rate results in the correct number of samples (note that 2) can only be guaranteed if the detector sampling rate is lower than the current sampling rate)
+                        2 * np.ceil(
                             detector.get_number_of_samples(station.get_id(), channel.get_id()) / 2
-                            * channel.get_sampling_rate() / detector_sampling_rate
+                            * sampling_rate / detector_sampling_rate
                         ))
+
                     if number_of_samples > trace.shape[0]:
                         logger.error("Input has fewer samples than desired output. Channels has only {} samples but {} samples are requested.".format(
                             trace.shape[0], number_of_samples))
                         raise AttributeError
                     else:
-                        sampling_rate = channel.get_sampling_rate()
-                        self.__check_sampling_rates(detector_sampling_rate, sampling_rate)
                         trigger_time_sample = int(np.round(trigger_time_channel * sampling_rate))
-                        # logger.debug(f"channel {channel.get_id()}: trace_start_time = {channel.get_trace_start_time():.1f}ns, trigger time channel {trigger_time_channel/units.ns:.1f}ns,  trigger time sample = {trigger_time_sample}")
-                        pre_trigger_time = self.__pre_trigger_time
+
                         channel_id = channel.get_id()
                         trigger_name = trigger.get_name()
-                        while isinstance(pre_trigger_time, dict):
-                            if trigger_name in pre_trigger_time.keys(): # keys are different triggers
-                                pre_trigger_time = pre_trigger_time[trigger_name]
-                            elif channel_id in pre_trigger_time: # keys are channel_ids
-                                pre_trigger_time = pre_trigger_time[channel_id]
-                            else:
+
+                        if isinstance(self.__pre_trigger_time, dict):
+                            if trigger_name not in self.__pre_trigger_time or channel_id not in self.__pre_trigger_time[trigger_name]:
                                 logger.error(
                                     'pre_trigger_time was specified as a dictionary, '
-                                    f'but neither the trigger_name {trigger_name} '
-                                    f'nor the channel id {channel_id} are present as keys'
-                                    )
+                                    f'but the trigger_name {trigger_name} or channel id {channel_id} '
+                                    'are not present as keys')
                                 raise KeyError
+
+                            pre_trigger_time = self.__pre_trigger_time[trigger_name][channel_id]
```

Comment:
Looks like this refactor will fail... `pre_trigger_time` can be a dict of "trigger names" but it also can be a dict of channel numbers... i.e. not a nested dict with [trigger_name][channel_id]

I just would not improvify this block

Reply:
autsch

okay